### PR TITLE
Factor code to download OmniSharp out of gulpfile.js

### PR DESCRIPTION
**Pull request by DustinCampbell**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -84,37 +49,37 @@ gulp.task('tslint', function () {
...
+	_fixes['./bin/omnisharp'] = '#!/bin/bash\n'
++ 'SOURCE="${BASH_SOURCE[0]}"\n'
++ 'while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink\n'
++ 'DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"\n'
++ 'SOURCE="$(readlink "$SOURCE")"\n'
++ '[[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located\n'
++ 'done\n'
++ 'DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"\n'
++ 'export SET DNX_APPBASE="$DIR/packages/OmniSharp/1.0.0/root"\n'
++ 'export PATH=/usr/local/bin:/Library/Frameworks/Mono.framework/Commands:$PATH # this is required for the users of the Homebrew Mono package\n'
```
Comment (gregg-miskelly) [inline gulpfile.js]: Do you still want this line?

Comment (gregg-miskelly) [inline gulpfile.js]: Nevermind, I guess this code is still using mono dnx.

**Comment by gregg-miskelly** [comment]
:+1:

File: gulpfile.js (hunk 1 above)
Comment (DustinCampbell) [inline gulpfile.js]: It's needed until I roll us forward to the new OmniSharp.

**Pull request merged**